File: src/test/java/com/comcast/viper/hlsparserj/tags/UnparsedTagTest.java
package com.comcast.viper.hlsparserj.tags;

import org.junit.Assert;
import org.junit.Test;

public class UnparsedTagTest {
	@Test
	public void testParsingNoValues() {
		UnparsedTag tag = new UnparsedTag("#EXTM3U");

		Assert.assertTrue(tag.getAttributes().isEmpty());
	}

	@Test
	public void testParsingSingleValue() {
		UnparsedTag tag = new UnparsedTag("#EXT-X-MEDIA-SEQUENCE:0");

		Assert.assertEquals("0", tag.getAttributes().get(Tag.UNNAMEDATTR0));
	}

	@Test
	public void testParsingDoubleValue() {
		UnparsedTag tag = new UnparsedTag("#EXTINF:100,Segment Title");

		Assert.assertEquals("100", tag.getAttributes().get(Tag.UNNAMEDATTR0));
		Assert.assertEquals("Segment Title", tag.getAttributes().get(Tag.UNNAMEDATTR1));
	}

	@Test
	public void testParsingAttributeWithEqualsSign() {
		UnparsedTag tag = new UnparsedTag("#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"group\",NAME=\"en\",URI=\"https://example.org/playlist.m3u8?param=value\"");

		Assert.assertEquals("https://example.org/playlist.m3u8?param=value", tag.getAttributes().get("URI"));
	}

	@Test
	public void testParsingAttributeWithoutEqualsSign() {
		UnparsedTag tag = new UnparsedTag("#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"group\",NAME=\"en\",URI=\"https://example.org/playlist.m3u8\"");

		Assert.assertEquals("https://example.org/playlist.m3u8", tag.getAttributes().get("URI"));
	}
	@Test
	public void testParsingAttributeWithComma() {
		UnparsedTag tag = new UnparsedTag("#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"group, subgroup\",NAME=\"en\",URI=\"https://example.org/playlist.m3u8\"");

		Assert.assertEquals("group, subgroup", tag.getAttributes().get("GROUP-ID"));
		Assert.assertEquals("https://example.org/playlist.m3u8", tag.getAttributes().get("URI"));
	}
}

File: src/test/kotlin/com/realeyes/hlsparserj/tags/UnparsedTagTest.kt
/**
*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.realeyes.hlsparserj.tags

import org.junit.Assert
import org.junit.Test

class UnparsedTagTest {
    @Test
    fun testParsingNoValues() {
        val tag = UnparsedTag("#EXTM3U")
        Assert.assertTrue(tag.getAttributes().isEmpty())
    }

    @Test
    fun testParsingSingleValue() {
        val tag = UnparsedTag("#EXT-X-MEDIA-SEQUENCE:0")
        Assert.assertEquals("0", tag.getAttributes()[Tag.UNNAMEDATTR0])
    }

    @Test
    fun testParsingDoubleValue() {
        val tag = UnparsedTag("#EXTINF:100,Segment Title")
        Assert.assertEquals("100", tag.getAttributes()[Tag.UNNAMEDATTR0])
        Assert.assertEquals("Segment Title", tag.getAttributes()[Tag.UNNAMEDATTR1])
    }

    @Test
    fun testParsingAttributeWithEqualsSign() {
        val tag =
            UnparsedTag("#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"group\",NAME=\"en\",URI=\"https://example.org/playlist.m3u8?param=value\"")
        Assert.assertEquals("https://example.org/playlist.m3u8?param=value", tag.getAttributes()["URI"])
    }

    @Test
    fun testParsingAttributeWithoutEqualsSign() {
        val tag =
            UnparsedTag("#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"group\",NAME=\"en\",URI=\"https://example.org/playlist.m3u8\"")
        Assert.assertEquals("https://example.org/playlist.m3u8", tag.getAttributes()["URI"])
    }

    @Test
    fun testParsingAttributeWithComma() {
        val tag =
            UnparsedTag("#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"group, subgroup\",NAME=\"en\",URI=\"https://example.org/playlist.m3u8\"")
        Assert.assertEquals("group, subgroup", tag.getAttributes()["GROUP-ID"])
        Assert.assertEquals("https://example.org/playlist.m3u8", tag.getAttributes()["URI"])
    }
}
